Nicer code, Better compression, Ability to call nbl.l() multiple times

tahajahangir opened the pull request.

tahajahangir committed:
Unobfuscated, compiled with google closure

tahajahangir committed:
jslinted

tahajahangir committed:
dont add same script twice, with multiple calls to nbl.l()

berklee commented:
Thanks for your contribution. Could you please give me some more background, as the diff just replaces all code wholesale?

tahajahangir commented:
First of all, I decided to change the NBL a bit (as in third commit), but i found the code unreadable (it was like a minfied or obfuscated code). It appears you had used short varialbe and function names to reduce the code size, but it can achieved by using a good minifier (like google clousre compiler).

At first commit, I only changed variable names (to meaningful ones) and also i converted object like structrue of NBL to function. So NBL is now a function with local variables rather than an object with multiple properties.
This is more common and more readable, and also leads to a smaller compacted file.

At second commit, i changed the NBL to be jslint compliant. There are some small changes like converting "a && b()" to "if (a){ b();}". Note that google closure converted it back to "a && b()" at compile time ;)

The third commit, is my goal to change NBL. The problem is that multiple calls to "nbl.l(['myscript.js'])" will add "myscript.js" to &lt;head> multiple times. I used NBL at a situation like this:

``` javascript
document.getElementById('showCalendar').onclick = function(){
    // Try to load calendar scripts files
    nbl.l([ ['calendar_script.js'], function() {
            // script is loaded
            showCalendar();
        }]);
};
```

I excpected that 'calendar_script.js' added to &lt;head> once, but the new &lt;script> is added with every call to nbl.l

tahajahangir committed:
README updated to reflect version 3

Commit:
New syntax for NBL 3

Commit:
Merge branch 'master' of github.com:tahajahangir/nbl

Commit:
New syntax for NBL 3

Commit:
New syntax for NBL 3

Commit:
Bugfix when scripts has query string part in address

tahajahangir committed:
Bugfix in unsettings onreadystatechange in IE6

tahajahangir committed:
Added safe files (disable loading scripts via data attributes)
